test-utils/src/main/java/com/criteo/publisher/mock/MockedDependenciesRule.java

```diff
@@ -132,6 +131,9 @@ public void evaluate() throws Throwable {
           throwable = t;
         } finally {
           try {
+            resetActivityLifecycleCallbacks();
+            waitForIdleState();
```

Comment:
As discussed could you please either move the line to a separate self-documented method or write a comment?

Reply:
done